src/Sylius/Bundle/AdminBundle/EventListener/OrderShowListener.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\EventListener;
+
+use Sylius\Bundle\ResourceBundle\Event\ResourceControllerEvent;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\OrderCheckoutStates;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+final class OrderShowListener
+{
+    public function showOrder(ResourceControllerEvent $event): void
+    {
+        /** @var OrderInterface $order */
+        $order = $event->getSubject();
```

Comment:
What about using an `Assert` instead here ?

Reply:
Yes, this makes much more sense because otherwise it will crash on the next line anyways, when it is not an order.